clients/src/main/java/org/apache/kafka/common/PartitionInfo.java

```diff
@@ -55,7 +57,7 @@ public Node leader() {
     /**
      * The complete set of replicas for this partition regardless of whether they are alive or up-to-date
      */
-    public Node[] replicas() {
+    public List<Node> replicas() {
```

Comment:
Ah, good point. I thought this class was only used internally, but it's actually exposed in KafkaConsumer and KafkaProducer. Well, that's annoying. Perhaps I'll just revert the change since it's not strictly necessary.